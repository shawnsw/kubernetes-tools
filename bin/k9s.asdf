#!/bin/bash

. ${HOME}/.kube_helper

# https://github.com/derailed/k9s/issues/240

export KUBECONFIG=${KUBECONFIG:-${HOME}/.kube/config}

${HOME}/.asdf/shims/k9s ${NAMESPACE} ${CONTEXT} "$@"
