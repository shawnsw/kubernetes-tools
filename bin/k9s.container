#!/bin/bash

K9S_IMAGE=${K9S_IMAGE:-quay.io/derailed/k9s}
KUBECONFIG=${KUBECONFIG:-${HOME}/.kube/config}
GCLOUD_CONFIG=${GCLOUD_CONFIG:-${HOME}/.config/gcloud}
K9SCONFIG=${HOME}/.k9s
SELINUX=${SELINUX:-":Z"}
GCLOUD=/usr/lib64/google-cloud-sdk/bin/gcloud
. ${HOME}/.kube_helper


MOUNT_K9SCONFIG=""
MOUNT_KUBECONFIG=""
MOUNT_GCLOUD=""

[ -f "${KUBECONFIG}" ] && MOUNT_KUBECONFIG="-v $(realpath ${KUBECONFIG}):/root/.kube/config${SELINUX}"
[ -d "${K9SCONFIG}" ] && MOUNT_K9SCONFIG="-v ${K9SCONFIG}:/root/.k9s${SELINUX}"
# [ -f "${GCLOUD}" ] && MOUNT_GCLOUD="-v ${GCLOUD}:${GCLOUD}${SELINUX}"
[ -d "${GCLOUD_CONFIG}" ] && MOUNT_GCLOUD="${MOUNT_GCLOUD} -v ${GCLOUD_CONFIG}:/root/.config/gcloud${SELINUX}"

podman run --rm --network=host -it ${MOUNT_KUBECONFIG} ${MOUNT_K9SCONFOG} ${MOUNT_GCLOUD} ${K9S_IMAGE} ${NAMESPACE} ${CONTEXT} $@
