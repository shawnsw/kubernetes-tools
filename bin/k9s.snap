#!/bin/bash

. ${HOME}/.kube_helper

# https://github.com/derailed/k9s/issues/240

export KUBECONFIG=${KUBECONFIG:-${HOME}/.kube/config}

${HOME}/.local/bin/k9s ${NAMESPACE} ${CONTEXT} "$@"
